File: go_server/internal/models/film.go
package models

type Film struct {
	ID          uint        `json:"id" gorm:"primaryKey"`
	Title       string      `json:"title" gorm:"not null"`
	Year        uint        `json:"year"`
	Country     string      `json:"country"`
	Genre       string      `json:"genre"`
	ImagePath   string      `json:"imagePath"`
	Actors      []*Actor    `json:"actors" gorm:"many2many:film_actors"`
	Categories  []*Category `json:"categories" gorm:"many2many:film_categories"`
	Description string      `json:"description" gorm:"type:text"`
}

File: go_server/static/js/film.js
var film = {};

function renderCurrentFilm() {
    console.log(film);   
    console.log($(".poster img"))
    $(".poster img").attr("src", film.imagePath);
    $(".film-title").text(film.title);
    $(".film-meta").text(film.country + "," + film.year + "·" + film.genre)
    $(".film-description").text(film.description)
}

function getFilmData(film_id) {
    console.log(film_id)
    $.post("/api/get-current-film", {"film_id" : film_id}, function(response) {
        film = response;  
        renderCurrentFilm();
    });
}

function main() {
    let params = new URLSearchParams(window.location.search);
    film_id = params.get("film_id");
    getFilmData(film_id);
}

main();
